src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -420,9 +432,25 @@ private static boolean isRoleAssignmentsOfInterestToAccessProcessCheck(RoleAssig
             || roleAssignment.getGrantType().equals(GrantType.CHALLENGED.name());
     }
 
-    private List<GrantType> generatePostFilteringAccessGrants(List<RoleAssignment> roleAssignments) {
+    private static boolean isRoleAssignmentsOfInterestToAccessGrantCheck(RoleAssignment roleAssignment) {
+        return isRoleAssignmentsOfInterestToAccessProcessCheck(roleAssignment)
+            || roleAssignment.getGrantType().equals(GrantType.BASIC.name());
+    }
+
+    private List<GrantType> generatePostFilteringAccessGrants(CaseDetails caseDetails,
+                                                              List<RoleAssignment> roleAssignments) {
+        String caseState = caseDetails.getState();
+        CaseTypeDefinition caseType = caseDefinitionRepository.getCaseType(caseDetails.getCaseTypeId());
+
         return roleAssignments.stream()
+            .filter(DefaultCaseDataAccessControl::isRoleAssignmentsOfInterestToAccessGrantCheck)
+            .filter(roleAssignment -> userHasAccessToCaseUsingRoleAssignments(caseState,
+                                                                              caseType,
+                                                                              List.of(roleAssignment))
+            )
             .map(roleAssignment -> GrantType.valueOf(roleAssignment.getGrantType()))
+            .distinct()
+            .sorted()
             .collect(Collectors.toList());
```

Comment:
Optional suggestion, as I appreciate it requires a little change to `CaseAccessMetadata` and any associated tests too:
```suggestion
            .collect(Collectors.toSet());
```
We always end up needing the distinct values and there's no requirement for ordering so a `Set` is a bit more appropriate, and we could also remove the (repeated) `.distinct().sorted()` from the getter in `CaseAccessMetadata`.

Reply:
Updated in f5f3be70b4f9421e46f4f2a96440eb99942b936c after discussion.  Calls to `distinct()` and `sorted()` removed as they are also performed in `CaseAccessMetadata.getAccessGrantsString(...)`, i.e. priror to output generation.  However left as list as Sort will make it easier to verify the CSV output in FTAs and volume of list is always low.